hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -281,15 +269,11 @@ private void scanInternalV1(Option<KeySpec> keySpecOpt) {
             LOG.info("Reading a data block from file " + logFile.getPath() + " at instant " + instantTime);
             // store the current block
             currentInstantLogBlocks.push(logBlock);
-            validLogBlockInstants.add(logBlock);
```

Comment:
Previously there was the logic below before `currentInstantLogBlocks.push(logBlock);` (see #9545):
```
LOG.info("Reading a data block from file " + logFile.getPath() + " at instant "
                + logBlock.getLogBlockHeader().get(INSTANT_TIME));
            if (isNewInstantBlock(logBlock) && !readBlocksLazily) {
              // If this is an avro data block belonging to a different commit/instant,
              // then merge the last blocks and records into the main result
              processQueuedBlocksForInstant(currentInstantLogBlocks, scannedLogFiles.size(), keySpecOpt);
            }
```
Is that still needed for branch-0.x?  Could you also raise a PR against `branch-0.x` if the logic is different between 1.x and 0.x release line?

Reply:
readBlocksLazily is always true. So, this was no-op code lying around.

Reply:
Got it.  I think it's better to put such cleanup changes in a separate PR next time so it's easier to review.

Reply:
sg. noted.